js/test/Exchange/test.sharedMethods.js

```diff
@@ -0,0 +1,206 @@
+    } else {
+        // if index was provided (mostly from fetchOHLCV) then we check if it exists, as mandatory
+        assert (!(entry[keyName] === undefined), 'timestamp index ' + keyName + ' is undefined' + logText);
+    }
+    const ts = entry[keyName];
+    if (ts !== undefined) {
+        assert (exchange.isInteger (ts), 'timestamp is not an integer' + logText);
+        assert (ts > 1230940800000, 'timestamp is impossible to be before 1230940800000 / 03.01.2009' + logText); // 03 Jan 2009 - first block
+        assert (ts < 2147483648000, 'timestamp more than 2147483648000 / 19.01.2038' + logText); // 19 Jan 2038 - int32 overflows // 7258118400000  -> Jan 1 2200
+        if (nowToCheck !== undefined) {
+            assert (ts < nowToCheck + 60000, 'trade timestamp is not below current time. Returned datetime: ' + exchange.iso8601 (ts) + ', now: ' + exchange.iso8601 (nowToCheck) + logText);
+        }
+    }
+    // only in case if the entry is a dictionary, thus it must have 'timestamp' & 'datetime' string keys
+    if (isDateTimeObject) {
+        // we also test 'datetime' here because it's certain sibling of 'timestamp'
+        assert (('datetime' in entry), 'datetime is missing from structure' + logText);
+        const dt = entry['datetime'];
+        if (dt !== undefined) {
+            assert (typeof dt === 'string', 'datetime is not a string' + logText);
+            assert (dt === exchange.iso8601 (entry['timestamp']), 'datetime is not iso8601 of timestamp' + logText);
+        }
+    }
+}
+
+
+function reviseCurrencyCode (exchange, method, entry, actualCode, expectedCode = undefined) {
+    const logText = logTemplate (exchange, method, entry);
+    if (actualCode !== undefined) {
+        assert (typeof actualCode === 'string', 'currency code should be either undefined or a string' + logText);
+        assert ((actualCode in exchange.currencies), 'currency code should be present in exchange.currencies' + logText);
+        if (expectedCode !== undefined) {
+            assert (actualCode === expectedCode, 'currency code in response (' + actualCode + ') should be equal to expected code (' + expectedCode + ')' + logText);
+        }
+    }
+}
+
+function reviseSymbol (exchange, method, entry, key, expectedSymbol = undefined) {
+    const logText = logTemplate (exchange, method, entry);
+    const actualSymbol = exchange.safeString (entry, key);
+    if (actualSymbol !== undefined) {
+        assert (typeof actualSymbol === 'string', 'symbol should be either undefined or a string' + logText);
+        assert ((actualSymbol in exchange.markets), 'symbol should be present in exchange.symbols' + logText);
+    }
+    if (expectedSymbol !== undefined) {
+        assert (actualSymbol === expectedSymbol, 'symbol in response (' + actualSymbol + ') should be equal to expected symbol (' + expectedSymbol + ')' + logText);
+    }
+}
+
+function Gt (exchange, method, entry, key, compareTo) {
```

Comment:
I would probably rename it to something more meaningful like `assertGreater`

Reply:
sure! again, the names were just spontaneously chosen.